### ISSUE TITLE
Wxmaxima splits the command line up when running a demo file.

### ISSUE THREAD

**Issue by richardgobeli** [andrejv/wxmaxima#945]
![capture](https://user-images.githubusercontent.com/31253976/34223661-c1fe794e-e585-11e7-86a9-a0a68d6a602e.PNG)

It does not matter which demo is run the command lines get chopped up and are hard to read.
This is from the linearalgebra demo file.   This did not run at first, because the extension of the file is .demo and should only be .dem

Macsyma had a way to  interrupt a demo file at the prompt, so that you could look at results and then continue with the demo at the next command in the demo file.  Does this version of Wxmaxima allow for this?

Thanks 
Richard Gobeli

**Comment by gunterkoenigsmann** [comment]
Continuing should be possible by pressing Ctrl+Enter or Shift+Enter. ...and the display issue should no more exist in the current nightly build of wxMaxima. Kind regards,

 Gunter-

**Comment by gunterkoenigsmann** [comment]
Ok. ...I've renamed the .demo file to .dem now in the maxima repository. Thanks for finding this!

**Comment by richardgobeli** [comment]
Yes,  I know about the shift-enter to continue the demo.   What I wanted to know is there is a way to interrupt the demo and do other calculations.  Then later to continue the demo at the next command in the demo.  At the prompt the math processor is locked and will not let me do a display command or any other command.

**Comment by gunterkoenigsmann** [comment]
I don't know about any such method. But as one can always open a separate window for a demo I don't know if this functionality is important anymore. If you feel it is: wxMaxima is merely the program that displays the maths. The program that actually runs the demo can be found at http://maxima.sourceforge.net

**Comment by richardgobeli** [comment]
Was the fix for the chopped demo source lines committed?  I tried the nightly build and did not see a change.

**Comment by richardgobeli** [comment]
It is a bit better.  It does still split some lines.  lineralgebra.dem below:
![image](https://user-images.githubusercontent.com/31253976/34320488-f9183c20-e7c0-11e7-82bb-ac4b14a57d66.png)

After installing the link to the maxima help is broken.  How do I fix it?

Richard

**Comment by gunterkoenigsmann** [comment]
In the code of wxMaxima there should be no line that is able to split a command in the middle of a word. At least I didn't find such a line. But Maxima does contain such a functionality. Which value is the variable linewidth set to in your Maxima?

Do you mean the link in the windows start menu or the one in the help menu of wxMaxima? And which path did you install the program into? 

Kind regards

 Gunter.

**Comment by richardgobeli** [comment]
Isn't the linewidth used for plotting?  My linel is set to default of 79.
I don't think the line character count gets set back to zero when running in demo mode.
All the quoted text prints OK.  Issue is only with the input command lines.  It seems to do it only once per command now, but not every input line.
Here is demo(tensor)
E for examples and the 11 for deriving Schwarzschild's metric

![capture](https://user-images.githubusercontent.com/31253976/34375826-2abd4578-eaae-11e7-8549-42687bdb33cd.PNG)


When I installed the nightly build, I let it go to the maxima-current folder.
The toolbar maxima help was not found.
The program could not find the maxima help file.  After I called some help items using ? name and restarting, it can now open the maxima help file from the toolbar.

Where is the best place to put the nightly build when installing?

One thing I noticed is that it does not define the correct app location.
![capture](https://user-images.githubusercontent.com/31253976/34375352-b6bb9b7c-eaab-11e7-9fa0-117d592b7008.PNG)
This stops me from setting wxmaxima as the default program based on file extension or adding Wxmaxima to the open with list.

**Comment by gunterkoenigsmann** [comment]
The linewidth is used for plotting, that is right. But there was a variable in maxima that tells the program after how many chars to split the line...

In the meantime a second user has reported this issue to me...
and after searching in the wrong place for what feels like weeks I have now found the feature in maxima:

     ? linel;

My hope is now that you not knowing about this obscure feature accidentally used linel as the name of a variable.

**Comment by richardgobeli** [comment]
I am not setting any variables.  I start with a clean wxmaxima and run any demo file.
Here I run demo(Eigen).  On line %i18 it does this:
![capture](https://user-images.githubusercontent.com/31253976/34524274-7f68db70-f060-11e7-9ef5-e942ec169697.PNG)
The Maxima response is this:
![capture](https://user-images.githubusercontent.com/31253976/34524297-a58ba5a8-f060-11e7-9d1d-6f5eb8435e8e.PNG)
At the end of the demo, I check linel;

![capture](https://user-images.githubusercontent.com/31253976/34524431-2086ab36-f061-11e7-919d-a54a4ac7c755.PNG)

**Comment by gunterkoenigsmann** [comment]
That would have been both too easy and too improbable at the same time, anyway.

If you enable the XML monitor (for example by double-clicking on the network symbol): Do you see the superfluous line breaks there, too? If this is true that would mean that we have to search for the problem on the maxima side, still.

**Comment by richardgobeli** [comment]
I show the raw maxiam monitor above.  It shows the command response is one line.

**Comment by gunterkoenigsmann** [comment]
Hmmm... ...that all seems to lead into one direction:

- Maxima sometimes sends messages to wxMaxima one char at a time and
- after all the speed ups wxMaxima now is able to process and to display them in small portions.

This all would explain why Roland tells that he has the impression that the program has gone slower: It now needs to do more each time it does do anything.

I will add a small delay to processing data from maxima today just for finding out if this theory is right. But if it were right - how would a real solution look? I could wait for a closing XML tag or for a newline or for 100ms every time I retrieve data from maxima. But that is a workaround, not a solution.

**Comment by richardgobeli** [comment]
I have this happen a lot when doing demos.   Wxmaxima just seems to stop and is waiting for something:
![capture1](https://user-images.githubusercontent.com/31253976/34534656-a1be51d6-f084-11e7-9752-d2ef160e8129.PNG)
![capture](https://user-images.githubusercontent.com/31253976/34534670-a9fa2c8a-f084-11e7-95e4-0d10caac9ce2.PNG)

**Comment by gunterkoenigsmann** [comment]
Ok... ...then the code that converts the data from maxima back to unicode is broken - which might explain the spurious newlines, too. Will have a look at this tomorrow.

**Comment by richardgobeli** [comment]
Should I make a new issue when things like this happen?

**Comment by gunterkoenigsmann** [comment]
Having separate tickets for separate problems makes things easier to track. I cannot promise to always have the manpower needed to resolve all issues as fast as I want, though...

**Comment by richardgobeli** [comment]
The last change with the delay made a big improvement.  The lines don't get split any more and Wxmaxima stalls about once in about 100 commands.

**Comment by gunterkoenigsmann** [comment]
Once in 100 commands is way too much. But it seems we at least know now in which direction to move...

**Comment by gunterkoenigsmann** [comment]
Actually once is too much in this respect.

**Comment by richardgobeli** [comment]
With this build the split lines are back.  The amount of stalling is still about the same.  I am lucky if I can get through some of the longer demo files with it not stalling.

**Comment by gunterkoenigsmann** [comment]
Ok... ...will try something different.

**Comment by gunterkoenigsmann** [comment]
I have now commented out my Unicode enhancements.

**Comment by richardgobeli** [comment]
I doesn't look like it is the Unicode enhancements.  It hangs up about as much as before and it still splits lines.

**Comment by richardgobeli** [comment]
This one really got screwed up.  I get errors like below:
![capture](https://user-images.githubusercontent.com/31253976/34645180-a00ab62e-f33e-11e7-8b9a-ad13ebc04f69.PNG)
![capture1](https://user-images.githubusercontent.com/31253976/34645181-a4041518-f33e-11e7-8bbc-99536492a7b0.PNG)

**Comment by richardgobeli** [comment]
The build from 1/8/2018 has the autocomplete working and the program did not stall after a section of going through the tensor demo about 4 times.  I have generally never gotten through it more than once.

It looks like the stalling has stopped.   The lines being split in the demo is still there.
lines are OK when not running a demo.

**Comment by richardgobeli** [comment]
Are you still going to be looking at this issue?
Thanks again for your help.

**Comment by gunterkoenigsmann** [comment]
I am. But I am quite unsure on what to do:
- I could keep a pointer to the last cell we got text for and append to this cell id the line continues. But If a line begins with "Err" and continues with "or" we won't see "error" in this case.
- I could wait until we get a full line (including newline) or an XML tag. But I cannot guarantee that a line will always end
- or we could wait 100 Milliseconds for the line end, or a second or... ...which (depending on the system load) can still go wrong
=> No optimum idea, still.

If I could at least reproduce the problem on my computer perhaps I could ask the maxima mailing list about this problem. That means:

What is the output of

       build_info();

for you?

Thanks a lot,
and kind regards,
   Gunter.

**Comment by richardgobeli** [comment]
![capture](https://user-images.githubusercontent.com/31253976/34790151-c90b52b6-f638-11e7-8ca6-2e6472e06a7e.PNG)

**Comment by gunterkoenigsmann** [comment]
Ok... ...I am always using sbcl (add

     -l sbcl

to the command-line options wxMaxima uses for maxima: Not that the error only happens if clisp is used as compiler for maxima...

**Comment by richardgobeli** [comment]
I added that to the command line.  The package is very fast now.  The lines do not get split anymore.
Graphs pop up very fast.

This option is not explained anywhere.
I think this needs to be documented.  For me this bug has fixed this issue.  The bug for me can be closed.

Thanks,

**Comment by richardgobeli** [comment]
I have noticed the on my laptop it uses CLISP.   If the -l is set wrong it will crash the program.  Shouldn't Wxmaxima or maxima automatically pick the lisp that it was compiled on?

**Comment by gunterkoenigsmann** [comment]
That is a Thing I stumbled upon, too. But that is something the guys at http://maxima.sourceforge.net have to resolve, not wxMaxima.

The original problem should be gone in the current nightly build, though.
